symengine/pow.cpp

```diff
@@ -164,6 +164,40 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         RCP<const Pow> A = rcp_static_cast<const Pow>(a);
         return pow(A->get_base(), mul(A->get_exp(), b));
     }
+    if (eq(*a, *E)) {
+        if (is_a<Mul>(*b)
+            and is_a_Complex(*down_cast<const Mul &>(*b).get_coef())) {
+            const map_basic_basic &dict = down_cast<const Mul &>(*b).get_dict();
+            RCP<const Number> coef = down_cast<const Mul &>(*b).get_coef();
+            if (dict.size() == 1) {
+                if ((is_a<Integer>(*mul(coef, I))
+                     or (is_a<Rational>(*mul(coef, I))
+                         and (eq(*mul(coef, I), *rational(1, 2))
+                              or eq(*mul(coef, I), *rational(-1, 2)))))) {
+                    for (const auto &p : dict) {
+                        if (eq(*p.first, *pi) and eq(*p.second, *one)) {
+                            return add(cos(mul(mul(I, minus_one), b)),
+                                       mul(I, sin(mul(mul(I, minus_one), b))));
```

Comment:
Since we only care about multiples of `I*pi/2`, how about checking for the multiple, and writing the formula for the 4 cases? Will be faster.

Reply:
To be clear, which four cases are we talking about?

Reply:
If the arg is `n*I*pi/2`, whether `n = 0,1,2,3 mod 4`